### DISCUSSION TITLE
Case Study: Mill vs Maven

### DISCUSSION THREAD

**Discussion by lefou** [netty/netty#14420] [Show and tell]
Hello Netty folks,

I just wanted to let you know, that there is a case study comparing two Build Tools for Java: Maven and Mill. In that document, The netty code base is ported to Mill to highlight the major similarities and differences.

https://mill-build.org/mill/0.12.1/comparisons/maven.html

> The Mill build for Netty is not 100% complete, but it covers most of the major parts of Netty: compiling Java, compiling and linking C code via JNI, running JUnit tests and some integration tests using H2Spec. All 47 Maven subprojects are modelled using Mill, with the entire Netty codebase being approximately 500,000 lines of code.

Best regards,

Tobias Roeser

**Comment by chrisvest** [comment]
The "comparing jar files" is revapi, which also causes the downloading of prior versions. Something that maven does not normally do. You can run our maven build with `-Pfast -DskipTests` to disable as much as possible. Add `-T1C` (or something like that) and maven will build as fast as it can go.

**Reply by lefou** [reply]
* [ ] https://github.com/com-lihaoyi/mill/issues/3871
